### DISCUSSION TITLE
Unable to get response in requested language completely

### DISCUSSION THREAD

**Discussion by z-u-a** [osm-search/Nominatim#3425] [Q&A] [answered]
Hello,

I have nominatim running in container. 

when I am giving "en-US" or just "en" in request to nominatim, it is giving _town_ in urdu language like following 
url: `"https://nominatim.******.net/nominatim/reverse?format=json&zoom=18&lat=24.85727850323225&lon=67.15176360660895&accept-language=es-US"`

response:
`{
  place_id: 935981,
  licence: "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  osm_type: "way",
  osm_id: 530765636,
  lat: "24.857273102405422",
  lon: "67.15175395599815",
  display_name: "Allah Buksh Goth, Bagh-e-Korangi, Landhi Town, لانڈھی, Korangi District, Sindh, 75120, Pakistan",
  address: {
    neighbourhood: "Allah Buksh Goth",
    suburb: "Bagh-e-Korangi",
    city_district: "Landhi Town",
    **town: "لانڈھی",**
    county: "Korangi District",
    state: "Sindh",
    "ISO3166-2-lvl4": "PK-SD",
    postcode: "75120",
    country: "Pakistan",
    country_code: "pk",
  },
  boundingbox: [
    "24.8571047",
    "24.8616883",
    "67.1491024",
    "67.1518482",
  ],
}`

but when I give any other locale for example de-DE, it gives _town_ name in english like following but gives some other content in urdu.

url: `'https://nominatim.********.net/nominatim/reverse?format=json&zoom=18&lat=24.85727850323225&lon=67.15176360660895&accept-language=de-DE'
`
response: 
` {
  place_id: 935981,
  licence: "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  osm_type: "way",
  osm_id: 530765636,
  lat: "24.857273102405422",
  lon: "67.15175395599815",
  display_name: "Allah Buksh Goth, باغ کورنگی, Landhi Town, Landhi, ضلع کورنگی, Sindh, 75120, Pakistan",
  address: {
    neighbourhood: "Allah Buksh Goth",
    suburb: "باغ کورنگی",
    city_district: "Landhi Town",
    **town: "Landhi",**
    county: "ضلع کورنگی",
    state: "Sindh",
    "ISO3166-2-lvl4": "PK-SD",
    postcode: "75120",
    country: "Pakistan",
    country_code: "pk",
  },
  boundingbox: [
    "24.8571047",
    "24.8616883",
    "67.1491024",
    "67.1518482",
  ],
}`

I need to have content completely either in the english or urdu language based on language headers. Need to understand the how nominatim handles the language. I tried to explore `reverse.php` but unable to find anything relevant.  TIA

Note: there is no as such middleware in my code which may transforming this

**Comment by mtmail** [comment, marked as answer]
The town has no english name in OpenStreetMap data https://www.openstreetmap.org/way/521782592

It's easy to add when you click the 'Edit' button on the page (https://www.openstreetmap.org/fixthemap). If you setup includes daily replication (https://nominatim.org/release-docs/latest/admin/Update/) then you'll see the change tomorrow.

**Reply by z-u-a** [reply]
Got it. Thanks
